### DISCUSSION TITLE
get_page_source() in opera

### DISCUSSION THREAD

**Discussion by slvamegestic** [seleniumbase/SeleniumBase#2559] [General]
I am trying to get the page source to know my proxy IP
below is the code.
```
        sb.driver.uc_open_with_reconnect("http://httpbin.org/ip",reconnect_time=10)
        sb.wait_for_ready_state_complete()
        sb.sleep(2)
        page_source = sb.get_page_source()
```
RESULT is
```
<html><head><meta name="color-scheme" content="light dark"></head><body><pre style="word-wrap: break-word; white-space: pre-wrap;">{
  "origin": "106.*****"
}
</pre></body></html>
```
The above result is correct and expected.Working on chrome browser

In opera, it gives opera's initial page HTML string. I'm sure the page is rendered correctly after only getting the page source. But why am I not getting the expected HTML string? Is this something anyone has experienced before? 

Does anyone have any advice regarding this?

Any another way to know my proxy IP?

**Comment by mdmintz** [comment]
It's working for me after I set the `binary_location` to the Opera executable. You could also try https://ipinfo.io/
